Lib/test/test_typing.py

```diff
@@ -3819,6 +3828,65 @@ def blah():
 
         blah()
 
+    def set_up_overloads(self):
+        def blah():
+            pass
+
+        overload1 = blah
+        overload(blah)
+
+        def blah():
+            pass
+
+        overload2 = blah
+        overload(blah)
+
+        def blah():
+            pass
+
+        return blah, [overload1, overload2]
+
+    def test_overload_registry(self):
+        impl, overloads = self.set_up_overloads()
+
+        self.assertEqual(list(get_overloads(impl)), overloads)
+        clear_overloads(impl)
+        self.assertEqual(get_overloads(impl), [])
+
+        impl, overloads = self.set_up_overloads()
+
+        self.assertEqual(list(get_overloads(impl)), overloads)
+        clear_overloads()
+        self.assertEqual(get_overloads(impl), [])
+
+    def test_variant_registry_repeated(self):
```

Comment:
The variant registry is dead; long live the overload registry

```suggestion
    def test_overload_registry_repeated(self):
```